extensions/notebook/src/dialog/addNotebookDialog.ts

```diff
@@ -0,0 +1,179 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as path from 'path';
+import { pathExists, remove } from 'fs-extra';
+import * as loc from '../common/localizedConstants';
+import { IconPathHelper } from '../common/iconHelper';
+import { FileType, IBookTocManager } from '../book/bookTocManager';
+import { confirmReplace, getDropdownValue } from '../common/utils';
+import { IPrompter } from '../prompts/question';
+import CodeAdapter from '../prompts/adapter';
+import { BookModel } from '../book/bookModel';
+import { BookTreeItemType } from '../book/bookTreeItem';
+
+export class AddNotebookDialog {
+	private dialog: azdata.window.Dialog;
+	public view: azdata.ModelView;
+	private formModel: azdata.FormContainer;
+	private notebookNameInputBox: azdata.InputBoxComponent;
+	private booksDropdown: azdata.DropDownComponent;
+	private notebookTypeDropdown: azdata.DropDownComponent;
+	private saveLocationInputBox: azdata.InputBoxComponent;
+	private prompter: IPrompter;
+	private bookNames: string[];
+
+	constructor(private tocManager: IBookTocManager, private books: BookModel[]) {
+		this.prompter = new CodeAdapter();
+		this.bookNames = this.books.length > 0 ? this.books.map(b => b.bookItems[0].title) : [];
+		this.bookNames.unshift('-');
+	}
+
+	protected createHorizontalContainer(view: azdata.ModelView, items: azdata.Component[]): azdata.FlexContainer {
+		return view.modelBuilder.flexContainer().withItems(items, { CSSStyles: { 'margin-right': '5px', 'margin-bottom': '10px' } }).withLayout({ flexFlow: 'row', alignItems: 'center' }).component();
+	}
+
+	public async selectFolder(): Promise<string | undefined> {
+		const allFilesFilter = loc.allFiles;
+		let filter: any = {};
+		filter[allFilesFilter] = '*';
+		let uris = await vscode.window.showOpenDialog({
+			filters: filter,
+			canSelectFiles: false,
+			canSelectMany: false,
+			canSelectFolders: true,
+			openLabel: loc.labelSelectFolder
+		});
+		if (uris && uris.length > 0) {
+			return uris[0].fsPath;
+		}
+		return undefined;
+	}
+
+	public async validatePath(folderPath: string, fileBasename: string): Promise<boolean> {
+		const destinationUri = path.join(folderPath, fileBasename);
+		if (await pathExists(destinationUri)) {
+			const doReplace = await confirmReplace(this.prompter, loc.confirmReplaceFile);
+			if (doReplace) {
+				//remove folder if exists
```

Comment:
You shouldn't need to delete it first - you can just write the new file over the old one.